### PR TITLE
Feature/13 file list component

### LINKED ISSUE #13: Add a file list component to the RupSetGenTask
We want to see the list of files linked to the task

so that we can see which files are produced/consumed to view the file details

and we want to get a download link 

NB:  file download links are expiring links to the resource in s3(default is 3600 seconds)

## UI Suggestions

Sorry, Sam I don't have time to do mock ups this week. 

## File List Component

each list item has: a) name, b) role and c) [get file] button
 - very long file names could be shortened e.g. 
    **RupSet_Az_FM(CFM_0_3_SANSTV.....mxSc(0.0).zip** in place of 
`RupSet_Az_FM(CFM_0_3_SANSTVZ)_mxSbScLn(0.5)_mxAzCh(60.0)_mxCmAzCh(560.0)_mxJpDs(5.0)_mxTtAzCh(60.0)_thFc(0.0).zip`  - this example is from the  PROD API
 - a button / component should fetch a new download link without leaving the current route. 
 - show if the file role READ, WRITE, READ/WRITE from the connection role attribute

### Query/Data examples

```
fragment rg0 on RuptureGenerationTask {
  id
  files {
    edges {
      node {
        ... on FileRelation {
          role
          file {
            ... on File {
              id
              file_name
            }
          }
        }
      }
    }
  }
}

query one_task {
  node(id: "UnVwdHVyZUdlbmVyYXRpb25UYXNrOjU5OXhISGhq") {
    ...rg0
  }
}```

returns 

{
  "data": {
    "node": {
      "id": "UnVwdHVyZUdlbmVyYXRpb25UYXNrOjU5OXhISGhq",
      "files": {
        "edges": [
          {
            "node": {
              "role": "WRITE",
              "file": {
                "id": "RmlsZToxMzM1LjAyNlBEeQ==",
                "file_name": "InversionSolution-CFM09_tf0.1-rnd0-t10.zip"
              }
            }
          },
          {
            "node": {
              "role": "WRITE",
              "file": {
                "id": "RmlsZToxMzM2LjB0Ulh2TQ==",
                "file_name": "java_app.25334.log"
              }
            }
          },
          {
            "node": {
              "role": "WRITE",
              "file": {
                "id": "RmlsZToxMzM3LjBOaHZUbw==",
                "file_name": "python_script.25334.log"
              }
            }
          }
        ]
      }
    }
  }
}```

```

### PR DESCRIPTION
**Pull request by wooneh**
Closes #13 

List of changes:

* Adds a file list to the RuptureGenerationTask view, with a link to the file.
* Polls for a new link after timeout expiry is reached.
* Truncates long text (mousing over shows full text).

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test --watchAll=false",
+    "test": "react-scripts test --watchAll=false --coverage",
```
Comment (chrisbc) [inline package.json:21]: nice touch :)

File: src/components/RuptureGenerationTask.tsx [hunk 2]
```diff
@@ -20,6 +24,20 @@ const ruptureGenerationTaskQuery = graphql`
...
         state
+        files {
+          edges {
+            node {
+              role
+              file {
+                ... on File {
+                  id
+                  file_name
+                  file_url
```
Comment (chrisbc) [inline src/components/RuptureGenerationTask.tsx:35]: FYI @wooneh this is a slightly simpler and slightly costlier approach than the one I'd described, due to the AWS roundtrip time associated with the the file_url  resolver. But it works and get us going so we can do the 'cheaper' one later if we need.

File: src/components/KeyValueTable.tsx [hunk 3]
```diff
@@ -45,7 +46,9 @@ const KeyValueTable: React.FC<KeyValueTableProps> = ({ header, data }: KeyValueT
           {data?.map((kv) => (
             <AlternatingRow key={kv?.k}>
               <TableCell className={classes.tableCell}>{kv?.k}</TableCell>
-              <TableCell className={classes.tableCell}>{kv?.v}</TableCell>
+              <TableCell className={classes.tableCell}>
+                <TruncateText text={kv?.v ?? ''} />
```
Comment (chrisbc) [inline src/components/KeyValueTable.tsx:50]: @wooneh this isn't where I want the truncation applied - sorry. For the K/V I think it best to show the whole thing as there's no easy way to see the full payload.

Where it **should** be used is the file_name column of the new FileList component. And it IS! just checked :).

**Review by chrisbc — CHANGES_REQUESTED** [review]
Looks good Sam, I like the code coverage :). Please just remove truncation from the KV table and then I'll merge.

File: src/components/RuptureGenerationTask.tsx (hunk 2 above)
Comment (wooneh) [reply src/components/RuptureGenerationTask.tsx:35]: Noted, thanks!

**Review by chrisbc — APPROVED** [review]

**Pull request merged**